### PR TITLE
BaSP-MR-98: Superadmin

### PR DESCRIPTION
**Pull request by victoriassol**
This PR is related to ticket: https://trello.com/c/7HZrj5tF/98-week-13-superadmin

- Implemented Redux in the resource in order to handle the components' states better.
- CRUD actions were updated.
- A new "Loader" component was added when the Superadmins list is being fetched.

### CODE REVIEW

File: src/Redux/SuperAdmins/thunks.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  editSuperadminError,
+  editSuperadminSuccess,
+  deleteSuperadminPending,
+  deleteSuperadminError,
+  deleteSuperadminSuccess,
+  resetPrimaryStates
+} from './actions';
+
+import { handleDisplayToast, setContentToast } from '../Shared/ResponseToast/actions';
+const responseInfo = { resState: '', resMessage: '', superadmins: [] };
```
Comment (francoax) [inline src/Redux/SuperAdmins/thunks.js]: I saw this and I searched if you are using it but no, only in the error actions, but after that payload is unused. I think you can remove this.

**Comment by francoax** [comment]
Everything works perfect !! Nice

**Review by francoax — APPROVED** [review]

File: src/Components/SuperAdmins/Form/index.jsx [hunk 2]
```diff
@@ -5,92 +5,21 @@ import Button from '../../Shared/Button';
...
-        openModal();
-      }
-    } catch (error) {
-      setResMessage('Failed to edit superadmin');
-      setState('fail');
-      openModal();
-    }
-  };
-
   const getItemById = async () => {
```
Comment (WalterArielPavicich) [inline src/Components/SuperAdmins/Form/index.jsx]: try-catch could be added

Comment (WalterArielPavicich) [reply src/Components/SuperAdmins/Form/index.jsx]: and I think this request should be moved to the "thunks" file

**Review by WalterArielPavicich — CHANGES_REQUESTED** [review]

File: src/Components/SuperAdmins/Form/index.jsx (hunk 2 above)
Comment (victoriassol) [reply src/Components/SuperAdmins/Form/index.jsx]: > and I think this request should be moved to the "thunks" file

I was not able to do so but I found another way of removing the request altogether!

Comment (WalterArielPavicich) [reply src/Components/SuperAdmins/Form/index.jsx]: You found a very good solution! Everything looks good and works perfect! Great job!

**Review by WalterArielPavicich — APPROVED** [review]
You found a very good solution! Everything looks good and works perfect! Great job!

**Review by JulianRiedinger7 — COMMENTED** [review]
All tests passed! Great work! Only remains to fix the delete functionality!

**Pull request merged**